container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.container.jdisc.state;
+
+import com.google.inject.Inject;
+import com.yahoo.collections.Tuple2;
+import com.yahoo.component.provider.ComponentRegistry;
+import com.yahoo.container.jdisc.state.StateHandler.JSONObjectWithLegibleException;
+import com.yahoo.jdisc.Request;
+import com.yahoo.jdisc.Response;
+import com.yahoo.jdisc.Timer;
+import com.yahoo.jdisc.handler.AbstractRequestHandler;
+import com.yahoo.jdisc.handler.ContentChannel;
+import com.yahoo.jdisc.handler.ResponseDispatch;
+import com.yahoo.jdisc.handler.ResponseHandler;
+import com.yahoo.jdisc.http.HttpHeaders;
+import com.yahoo.metrics.MetricsPresentationConfig;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static com.yahoo.container.jdisc.state.StateHandler.getSnapshotPreprocessor;
```

Comment:
Consider moving `getSnapshotPreprocessor ` to separate class

Reply:
If there were more than one method, I would have created a common utility class or something. ATM, it looks like overkill.